### PR TITLE
consuming cats api

### PR DESCRIPTION
**Pull request by JuanMiChee**

### CODE REVIEW

File: HarrisMoviesApplication/ViewController.swift [hunk 1]
```diff
@@ -8,13 +8,42 @@
...
+    @IBOutlet weak var catFactTableView: UITableView!
+    
+    override func viewDidLoad() {
+        let url = URL(string: "https://cat-fact.herokuapp.com/facts")!
+        
+            
+            let task = URLSession.shared.dataTask(with: url) { data, response, error in
+                if let data = data {
+                    if let facts = try? JSONDecoder().decode([CatFact].self, from: data) {
+                        print(facts)
```
Comment (angelcornershop) [inline HarrisMoviesApplication/ViewController.swift]: Here we need to assign the received array to our VC array
```suggestion
                        self.factsArray = facts
```

File: HarrisMoviesApplication/ViewController.swift [hunk 2]
```diff
@@ -8,13 +8,42 @@
...
+        // MARK: - Properties
 
+        let text: String
+    }
+    
+    let factsArray:[CatFact] = []
 
+    @IBOutlet weak var catFactTableView: UITableView!
+    
+    override func viewDidLoad() {
```
Comment (angelcornershop) [inline HarrisMoviesApplication/ViewController.swift:16]:
```suggestion
    override func viewDidLoad() {
    super.viewDidLoad()
```

File: HarrisMoviesApplication/ViewController.swift [hunk 3]
```diff
@@ -8,13 +8,42 @@
...
 
-    override func viewDidLoad() {
-        super.viewDidLoad()
-        // Do any additional setup after loading the view.
-        //creating
-    }
+        // MARK: - Properties
 
+        let text: String
+    }
```
Comment (angelcornershop) [inline HarrisMoviesApplication/ViewController.swift]: Lets move this struct to a separate file

File: HarrisMoviesApplication/ViewController.swift [hunk 4]
```diff
@@ -8,13 +8,42 @@
...
+                    print("HTTP Request Failed \(error)")
+                }
+                
+                
+            }
+        
+        task.resume()
+        
+        catFactTableView.delegate = self
+        catFactTableView.dataSource = self
```
Comment (angelcornershop) [inline HarrisMoviesApplication/ViewController.swift]: Move this to the beginning of the function as this is a setup method we need done before anything else

File: HarrisMoviesApplication/TableViewProtocols.swift [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+
+extension ViewController: UITableViewDelegate, UITableViewDataSource{
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return factsArray.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell =  catFactTableView.dequeueReusableCell(withIdentifier: "prototypeCell", for: indexPath)
+        //cell.textLabel?.text = vamo[indexPath.row] esto es lo que hay que arreglar pa
```
Comment (angelcornershop) [inline HarrisMoviesApplication/TableViewProtocols.swift]: What is "vamo" ?

**Review by angelcornershop — CHANGES_REQUESTED** [review]

File: HarrisMoviesApplication/ViewController.swift [hunk 6]
```diff
@@ -2,19 +2,42 @@
 //  ViewController.swift
 //  HarrisMoviesApplication
 //
-//  Created by juan harrington on 20/03/22.
+//  Created by Juan Harrington ✅ on 20/03/22.
```
Comment (angelbetancourt) [inline HarrisMoviesApplication/ViewController.swift:5]: update this in [Xcode settings](https://stackoverflow.com/questions/8650661/how-to-change-developer-name-company-name-in-xcode):

**Review by angelbetancourt — CHANGES_REQUESTED** [review]

File: HarrisMoviesApplication/ViewController.swift (hunk 1 above)
Comment (JuanMiChee) [reply HarrisMoviesApplication/ViewController.swift]: oh i forgot it thank you

File: HarrisMoviesApplication/ViewController.swift (hunk 2 above)
Comment (JuanMiChee) [reply HarrisMoviesApplication/ViewController.swift:16]: i already move it, you can see it in the last commit

File: HarrisMoviesApplication/ViewController.swift (hunk 3 above)
Comment (JuanMiChee) [reply HarrisMoviesApplication/ViewController.swift]: ok i will separate all my files efficiently from now

File: HarrisMoviesApplication/ViewController.swift (hunk 4 above)
Comment (JuanMiChee) [reply HarrisMoviesApplication/ViewController.swift]: why would i do that? isn't the same?

File: HarrisMoviesApplication/TableViewProtocols.swift (hunk 5 above)
Comment (JuanMiChee) [reply HarrisMoviesApplication/TableViewProtocols.swift]: was a testing name for a variable i already resolved it

File: HarrisMoviesApplication/ViewController.swift [hunk 7]
```diff
@@ -2,19 +2,42 @@
...
 //
 
 import UIKit
 
 class ViewController: UIViewController {
+    
+    var factsArray:[CatFact] = []
```
Comment (angelbetancourt) [inline HarrisMoviesApplication/ViewController.swift:12]:
```suggestion
    private var factsArray:[CatFact] = []
```

File: HarrisMoviesApplication/ViewController.swift (hunk 4 above)
Comment (angelbetancourt) [reply HarrisMoviesApplication/ViewController.swift]: Isn't the same - We need these properties to be set before we use the tableview later. The tableview doesn't work without the datasource

File: HarrisMoviesApplication/ViewController.swift (hunk 6 above)
Comment (angelbetancourt) [reply HarrisMoviesApplication/ViewController.swift:5]: This is missing

File: HarrisMoviesApplication/ViewController.swift (hunk 7 above)
Comment (JuanMiChee) [reply HarrisMoviesApplication/ViewController.swift:12]: 👍

File: HarrisMoviesApplication/ViewController.swift (hunk 4 above)
Comment (JuanMiChee) [reply HarrisMoviesApplication/ViewController.swift]: oh it makes sense, ill check it thank you

**Review by angelbetancourt — APPROVED** [review]
Remember to change the PR title:

**Pull request merged**